benchmarking/evaluations/metrics/average_solo_status.py

```diff
@@ -0,0 +1,51 @@
+from collections import Counter
+from typing import Dict, List
+
+from api.models.team_set import TeamSet
+from benchmarking.evaluations.interfaces import TeamSetMetric
+
+
+class AverageSoloStatus(TeamSetMetric):
+    """
+    Calculate the average number of students having solo status in a team. A student is only considered as solo status
+    once, even if they have multiple minority attributes.
+
+    Params
+    ------
+    minority_groups: Dict[int, List[int]]
+        A dictionary where the key is the attribute_id and the value is a list of attribute_values that are considered
+        as minority.
+    """
+
+    def __init__(self, minority_groups_map: Dict[int, List[int]], *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.minority_groups_map = minority_groups_map
+
+        if len(minority_groups_map) == 0:
+            raise ValueError("minority_groups cannot be empty")
+
+    def calculate(self, team_set: TeamSet) -> float:
+        total_solo_status_students = set()
+        total_students = 0.0
```

Comment:
nit;

is this supposed to be a count of all unique students?

if so, it is non-intuitive why it's defaulted to the float of 0, instead of just 0 (I know why, but still, burned an extra few minutes trying to understand this one choice)